tests/components/pi_hole/test_config_flow.py

```diff
@@ -100,3 +69,32 @@ async def test_flow_user_invalid(hass: HomeAssistant):
         assert result["type"] == FlowResultType.FORM
         assert result["step_id"] == "user"
         assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_reauth(hass: HomeAssistant):
+    """Test reauth flow."""
+    mocked_hole = _create_mocked_hole(has_data=False)
+    entry = MockConfigEntry(domain=pi_hole.DOMAIN, data=CONF_DATA_DEFAULTS)
+    entry.add_to_hass(hass)
+    with _patch_init_hole(mocked_hole), _patch_config_flow_hole(mocked_hole):
+        assert not await hass.config_entries.async_setup(entry.entry_id)
+
+        flows = hass.config_entries.flow.async_progress()
+
+        assert len(flows) == 1
+        assert flows[0]["step_id"] == "reauth_confirm"
+        assert flows[0]["context"]["entry_id"] == entry.entry_id
+
+        mocked_hole.data = ZERO_DATA
+
+        result = await hass.config_entries.flow.async_configure(
+            flows[0]["flow_id"],
+            user_input={CONF_API_KEY: "newkey"},
+        )
+
+        await hass.async_block_till_done()
+        assert result["type"] == FlowResultType.ABORT
+        assert result["reason"] == "reauth_successful"
+
+        entry = hass.config_entries.async_get_entry(entry.entry_id)
```

Comment:
We have access to `entry` from above. It's the same instance.

Reply:
[`05cb098`](https://github.com/home-assistant/core/pull/85264/commits/05cb098956fb06d9340fd918862e2b6fa277a4c7)